onnxruntime/core/providers/cpu/tensor/col2im_attributes.h

```diff
@@ -0,0 +1,50 @@
+/**
+* Copyright (c) 2016-present, Facebook, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+/* Modifications Copyright (c) Microsoft. */
+
+#pragma once
+
+#ifndef SHARED_PROVIDER
+#include "core/common/common.h"
+#include "core/providers/common.h"
+#include "core/util/math.h"
+#endif
+
+#include "core/common/inlined_containers.h"
+#include "core/framework/op_kernel.h"
+#include "core/framework/op_node_proto_helper.h"
+
+namespace onnxruntime {
+
+struct Col2ImAttributes {
+  using Col2ImPadVector = InlinedVector<int64_t, kTensorShapeSmallBufferElementsSize * 2>;
```

Comment:
why do we need to specify the InlinedVector size here? it should have a reasonable default

Reply:
Copied from onnxruntime/core/providers/cpu/nn/conv_attributes.h which has the same set of attrbiutes/defauls. e.g. `using ConvPadVector = InlinedVector<int64_t, kTensorShapeSmallBufferElementsSize * 2>;` is analogous to `using Col2ImPadVector = InlinedVector<int64_t, kTensorShapeSmallBufferElementsSize * 2>;`

Do you think this is ok to keep?

Reply:
I think unless there's a good reason to specify a size, `InlinedVector<int64_t>` is fine.

Reply:
> > I think unless there's a good reason to specify a size, `InlinedVector<int64_t>` is fine.
> 
> There is a good reason. `InlinedVector` fits 6 elements for `int64_t`. Our max shape length is 5 and convolution requires twice as much at max, i.e. 10. So we want at least 10 to prevent reallocations.

That is what the code is already doing, right? `kTensorShapeSmallBufferElementsSize * 2` which is `5 * 2 = 10`

Reply:
ok sounds good